mixer/adapter/dogstatsd/config/config.proto

```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+// The `dogstatsd` adapter is designed to deliver Istio metric instances to a
+// listening [DataDog](https://www.datadoghq.com/) agent.
+package adapter.dogstatsd.config;
+
+import "gogoproto/gogo.proto";
+
+option go_package="config";
+
+// Configuration parameter for the DataDog adapter. These params control how mixer telemetry is transformed and sent to a DataDog agent.
+//
+// The adapter assumes that a datadog daemon is running as a sidecar or at some other endpoint that the mixer can reach.
+// Any dimension that is a part of the metric is converted to a tag automatically. The configuration of the DataDog agent/daemon is outside the scope of the adapter.
+message Params {
+    // Address of the dogstatsd server. 
+    // Default: localhost:8125
+    string address = 1;
+
+    // prefix to prepend to all metrics handled by the adapter. Metric "bar" with prefix "foo." becomes "foo.bar" in DataDog.
+    // Default: ""
+    string prefix = 2;
+
+    // UDP buffer size in bytes 
```

Comment:
I took another look at the client and it turns out this buffer is totally different than what I thought it was. I thought it was for adjusting the UDP packet size to fill before sending metrics (similar to the non-datadog statsd client) but instead its a buffer of individual metrics regardless of size. The buffer is flushed every 100ms regardless of size and can result in packet drops if the buffer is large.

Given this I think a default of non-buffering makes more sense but I'm torn on leaving this as a knob for tuning (with a warning) given that it could lead to undesired behavior. Thoughts?